src/DropdownReferenceSet/components/DropdownReferenceSet.ts

```diff
@@ -0,0 +1,158 @@
+                className: classNames("widget-dropdown-reference-set"),
+                onClick: this.setDropdownSize
+            },
+                this.props.selectType === "normal"
+                    ? this.props.isReadOnly
+                        ? createElement("input", {
+                            type: "text",
+                            readonly: "readonly",
+                            className: "form-control",
+                            disabled: "disabled",
+                            value: this.processOptions()
+                        })
+                        : createElement(Select, {
+                            options: this.props.data,
+                            noResultsText: "",
+                            ...commonProps
+                        })
+                    : createElement(Async, {
+                        autoload: false,
+                        onCloseResetsInput: false,
+                        searchPromptText: this.props.minimumCharacter > 0
+                            ? `Type more than ${this.props.minimumCharacter} characters to search`
+                            : "Type to search",
+                        loadOptions: this.props.asyncData,
+                        ...commonProps
+                    }),
+                createElement(Alert, { className: "widget-dropdown-reference-set-alert" }, this.props.alertMessage)
+            );
+        } else {
+            return createElement("p", { className: classNames("form-control-static", "read-only-text") },
+                this.processOptions()
+            );
+        }
+    }
+
+    private setDropdownSize = () => {
+        const dropdown = document.getElementsByClassName("Select-menu-outer");
+        const dropdownElement = dropdown[0] as HTMLElement;
+        if (dropdownElement && dropdownElement.style.visibility !== "visible") {
+            const dropdownDimensions = dropdownElement.getBoundingClientRect();
+            if (dropdownElement && dropdownDimensions) {
+                dropdownElement.style.width = dropdownDimensions.width - .08 + "px";
+                dropdownElement.style.left = dropdownDimensions.left + "px";
+                dropdownElement.style.top = dropdownDimensions.top + "px";
+                dropdownElement.style.visibility = "visible";
+                dropdownElement.style.position = "fixed";
+            }
+        }
+    }
+
+    private createSelectorProp(): { placeholder?: string, value?: object } {
+        if (this.props.selectedValue.length > 0) {
+            return { value: this.props.selectedValue };
+        }
+
+        return { placeholder: this.props.emptyOptionCaption };
+    }
+
+    private processOptions() {
+        let selectedLabel = "";
+        let formatedOptions = [] as any;
+
+        if (this.props.selectedValue.length > 0) {
+            formatedOptions = this.props.selectedValue.map((selectedGuid: any) => {
+                if (this.props.selectedValue) {
```

Comment:
This line will always be true, because it's inside the map of itself! It needs to be merged with line 141.

Suggestion: remove the line 143 and at line 141 use: **`if (this.props.selectedValue && this.props.selectedValue.length > 0) {`**